python/GafferUI/SpreadsheetUI.py

```diff
@@ -1439,6 +1439,8 @@ def popupEditor( cls, plug, plugBound ) :
 					return
 
 		windowSize = cls.__currentWindow._qtWidget().sizeHint()
+
+		cls.__currentWindow.resizeToFitChild()
 		cls.__currentWindow.setPosition( plugBound.center() - imath.V2i( windowSize.width() / 2, windowSize.height() / 2 ) )
```

Comment:
If we're going to call `resizeToFitChild()` then we can also remove the cheeky call to `sizeHint()` at L1441 :

```suggestion
		cls.__currentWindow.resizeToFitChild()
		windowSize = cls.__currentWindow.bound().size()
		cls.__currentWindow.setPosition( plugBound.center() - windowSize / 2 )
```

I also took a look to see if we could make this "just work" without any changes to SpreadsheetUI. It seems that when the window hasn't been shown yet, the `frameGeometry()` will always be `640x480`. We could detect that case using `windowHandle() is None` in `__constrainToScreen()` and then request the true size via `sizeHint()`. I don't think that's the answer for the Spreadsheet case, because we need to know the size before making the `setPosition()` call anyway. But maybe it's worth considering for the general case? Then again, maybe best not to touch anything now we finally have it doing what we want!